plugins/google-sheets/src/pages/MapSheetFields.tsx

```diff
@@ -90,26 +79,35 @@ const inferFieldType = (cellValue: CellValue): CollectionFieldType => {
     return "string"
 }
 
-const getFieldType = (context: PluginContext, columnIndex: number, cellValue?: CellValue): CollectionFieldType => {
+const getFieldType = (context: PluginContext, columnId: string, cellValue?: CellValue): CollectionFieldType => {
     // Determine if the field type is already configured
     if ("collectionFields" in context) {
-        const field = context.collectionFields?.find(field => Number(field.id) === columnIndex)
+        const field = context.collectionFields?.find(field => field.id === columnId)
         return field?.type ?? "string"
     }
 
     // Otherwise, infer the field type from the cell value
     return cellValue ? inferFieldType(cellValue) : "string"
 }
 
-const createFieldConfig = (headerRow: HeaderRow, context: PluginContext, row?: Row): CollectionFieldConfig[] => {
-    return headerRow.map((headerName, columnIndex) => ({
-        field: {
-            id: String(columnIndex),
-            name: headerName,
-            type: getFieldType(context, columnIndex, row?.[columnIndex]),
-        } as ManagedCollectionField,
-        originalFieldName: headerName,
-    }))
+const createFieldConfig = (
+    headerRow: HeaderRow,
+    uniqueColumnNames: string[],
+    context: PluginContext,
+    row?: Row
+): CollectionFieldConfig[] => {
+    return headerRow.map((headerName, columnIndex) => {
+        const sanitizedName = uniqueColumnNames[columnIndex]
+
+        return {
+            field: {
+                id: sanitizedName,
+                name: sanitizedName,
+                type: getFieldType(context, sanitizedName, row?.[columnIndex]),
+            } as ManagedCollectionField,
+            originalFieldName: headerName,
```

Comment:
This is not used anymore.

Reply:
Hmm, why was it there in the first place? Can drop it, probably.